downloads/swan-lake-release-notes/swan-lake-preview4/RELEASE_NOTE.md

```diff
@@ -1,30 +1,490 @@
+        - [Class definitions](#class-definitions)
+        - [Object-constructor expressions](#object-constructor-expressions)
+        - [Distinct objects](#distinct-objects)
+        - [Final fields in objects](#final-fields-in-objects)
+        - [Isolated functions](#isolated-functions)
+        - [Improved support for quoted identifiers](#improved-support-for-quoted-identifiers)
+        -  [Subsequent initialization support for local final variables](#subsequent-initialization-support-for-local-final-variables)
+        - [Support to handle errors at a single point - (the `on fail` clause)](#support-to-handle-errors-at-a-single-point-the-on-fail-clause)
+        - [Fail statement](#fail-statement)
+        - [Improved lang library functions](#improved-lang-library-functions)
+    - [Runtime](#runtime)
+        - [jBallerina Runtime APIs](#jballerina-runtime-apis)
+        - [Asynchronous Java interoperability](#asynchronous-java-interoperability)
+    - [Standard Library](#standard-library)
+    - [Developer Tools](#developer-tools)
+        - [The OpenAPI Tool](#the-openapi-tool)
+        - [Improvements related to code actions](#improvements-related-to-code-actions)
 
-If you are already using jBallerina version 1.2.0, or above, you can directly update your distribution to jBallerina {{ site.data.stable-latest.metadata.version }} by executing the following command:
+### Updating Ballerina
 
-> $ ballerina dist update
+You can use the update tool to update to Ballerina Swan Lake Preview 4 as follows.
 
-However, you need to use the following commands instead of the above if you have installed:
+#### For existing users
 
-- jBallerina 1.2.0 but switched to a previous version: `$ ballerina dist pull jballerina-{{ site.data.stable-latest.metadata.version }}`
-- a jBallerina version below 1.1.0: install via the [installers](/downloads/)
+If you are already using Ballerina, you can directly update your distribution to the Swan Lake channel using the [Ballerina Update Tool](http://ballerina.io/swan-lake/learn/keeping-ballerina-up-to-date/). To do this, first, execute the command below to get the update tool updated to its latest version. 
+                        
+> `ballerina update`
 
-**For new users:**
+ Next, execute the command below to update to Swan Lake Preview 4.
 
-If you have not installed jBallerina, then download the [installers](/downloads/) to install.
+ > `ballerina dist pull slp4`                  
+
+However, if you are using a Ballerina version below 1.1.0, install via the [installers](https://ballerina.io/downloads/).
+
+#### For new users
+
+If you have not installed Ballerina, then download the [installers](https://ballerina.io/downloads/) to install.
+
+### Highlights
+
+- Revised Ballerina object syntax, which differentiates the object type from the `class`, from which you can create object values. 
+- Introducing isolated functions to achieve concurrency safety together with read only values.  
```

Comment:
Shouldn't it be read-only?

Reply:
Yeah it should be `readonly`

Reply:
The correct term is `read-only`. Will change.

Reply:
Done via https://github.com/ballerina-platform/ballerina-dev-website/pull/1311.